apps/directory/src/components/biobankcards-components/BiobankCard.vue

```diff
@@ -0,0 +1,383 @@
+                multi
+              ></collection-selector>
+            </div>
+            <hr class="mt-1" v-if="numberOfCollections" />
+            <div v-else class="pl-2">This biobank has no collections yet.</div>
+            <div
+              class="collection-items mx-1"
+              v-for="(collectionDetail, index) of biobank.collectionDetails"
+              :key="collectionDetail.id"
+            >
+              <div v-if="showCollections" class="mb-2">
+                <div class="pl-2 pt-2 d-flex">
+                  <router-link
+                    :to="'/collection/' + collectionDetail.id"
+                    title="Collection details"
+                    class="text-dark"
+                  >
+                    <span
+                      class="fa fa-server collection-icon fa-lg mr-2 text-primary"
+                      aria-hidden="true"
+                    ></span>
+                    <span class="collection-name">{{
+                      collectionDetail.name
+                    }}</span>
+                  </router-link>
+                  <div class="ml-auto">
+                    <collection-selector
+                      class="ml-auto"
+                      :biobankData="biobank"
+                      :collectionData="collectionDetail"
+                      iconOnly
+                      bookmark
+                    ></collection-selector>
+                  </div>
+                </div>
+
+                <small>
+                  <view-generator
+                    class="p-1"
+                    :viewmodel="collectionViewmodel(collectionDetail)"
+                  />
+
+                  <matches-on :viewmodel="collectionDetail" class="px-1 ml-1" />
+                  <router-link
+                    :to="'/collection/' + collectionDetail.id"
+                    :title="`${collectionDetail.name} details`"
+                    class="text-info ml-1 pl-1"
+                  >
+                    <span>More details</span>
+                  </router-link>
+                </small>
+                <hr v-if="index != lastCollection" />
+                <div v-else class="pb-3"></div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </section>
+  </article>
+</template>
+
+<script>
+import {
+  getBiobankDetails,
+  getCollectionDetails,
+} from "../../functions/viewmodelMapper";
+import ViewGenerator from "../generators/ViewGenerator.vue";
+import CollectionSelector from "../checkout-components/CollectionSelector.vue";
+import InfoPopover from "../popovers/InfoPopover.vue";
+import MatchesOn from "../biobankcards-components/MatchesOn.vue";
+import { useSettingsStore } from "../../stores/settingsStore";
+import { useBiobanksStore } from "../../stores/biobanksStore";
+import { useQualitiesStore } from "../../stores/qualitiesStore";
+import { useCheckoutStore } from "../../stores/checkoutStore";
+
+export default {
+  setup() {
```

Comment:
When adding these setup functions, wouldn't it be better to switch to the composition api instead of the options api? I feel like that's more readable and the linter is better at dealing with it.

Reply:
composition api gives certain racing condition bugs.